include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t              maxGeometryShaderInvocations;
+    //uint32_t              maxGeometryInputComponents;
+    //uint32_t              maxGeometryOutputComponents;
+    //uint32_t              maxGeometryOutputVertices;
+    //uint32_t              maxGeometryTotalOutputComponents;
+    //uint32_t              maxFragmentInputComponents;
+    //uint32_t              maxFragmentOutputAttachments;
+    //uint32_t              maxFragmentDualSrcAttachments;
+    //uint32_t              maxFragmentCombinedOutputResources;
+    uint32_t maxComputeSharedMemorySize;
+    //uint32_t              maxComputeWorkGroupCount[3];
+    //uint32_t              maxComputeWorkGroupInvocations;
+    uint32_t maxWorkgroupSize[3];
+    //uint32_t              subPixelPrecisionBits;
+    //uint32_t              subTexelPrecisionBits;
+    //uint32_t              mipmapPrecisionBits;
+    //uint32_t              maxDrawIndexedIndexValue;
+    uint32_t maxDrawIndirectCount;
+    //float                 maxSamplerLodBias;
+    float    maxSamplerAnisotropyLog2;
+    uint32_t maxViewports;
+    uint32_t maxViewportDims[2];
+    //float                 viewportBoundsRange[2];
+    //uint32_t              viewportSubPixelBits;
+    //size_t                minMemoryMapAlignment;
+    uint32_t bufferViewAlignment;
+    uint32_t UBOAlignment;
+    uint32_t SSBOAlignment;
+    //int32_t               minTexelOffset;
+    //uint32_t              maxTexelOffset;
+    //int32_t               minTexelGatherOffset;
+    //uint32_t              maxTexelGatherOffset;
+    //float                 minInterpolationOffset;
+    //float                 maxInterpolationOffset;
+    //uint32_t              subPixelInterpolationOffsetBits;
+    //uint32_t              maxFramebufferWidth;
+    //uint32_t              maxFramebufferHeight;
+    //uint32_t              maxFramebufferLayers;
+    //VkSampleCountFlags    framebufferColorSampleCounts;
+    //VkSampleCountFlags    framebufferDepthSampleCounts;
+    //VkSampleCountFlags    framebufferStencilSampleCounts;
+    //VkSampleCountFlags    framebufferNoAttachmentsSampleCounts;
+    //uint32_t              maxColorAttachments;
+    //VkSampleCountFlags    sampledImageColorSampleCounts;
+    //VkSampleCountFlags    sampledImageIntegerSampleCounts;
+    //VkSampleCountFlags    sampledImageDepthSampleCounts;
+    //VkSampleCountFlags    sampledImageStencilSampleCounts;
+    //VkSampleCountFlags    storageImageSampleCounts;
+    //uint32_t              maxSampleMaskWords;
+    //VkBool32              timestampComputeAndGraphics;
+    float    timestampPeriodInNanoSeconds; // timestampPeriod is the number of nanoseconds required for a timestamp query to be incremented by 1 (a float because vulkan reports), use core::rational in the future
+    //uint32_t              maxClipDistances;
+    //uint32_t              maxCullDistances;
+    //uint32_t              maxCombinedClipAndCullDistances;
```

Comment:
We want to expose these

Reply:
`GL_MAX_CLIP_PLANES`  is in basic GL and GLES 

`GL_MAX_CULL_DISTANCES` and `GL_MAX_COMBINED_CLIP_AND_CULL_DISTANCES` are for Clipping distances but only provided by `EXT_clip_cull_distance` or `ARB_cull_distance` extension

Reply:
Done,
1. again, how did you figure these out? was it experience or do you have a process of checking these?
2. we have `GL_MAX_CLIP_DISTANCES` and not `GL_MAX_CLIP_PLANES` in our gl headers so I used that

Reply:
> 1. again, how did you figure these out? was it experience or do you have a process of checking these?

clip distances are ancient, so I guess experience.

The "cull distance" stuff is new to me, so I went looking and found the extensions

> 2. we have GL_MAX_CLIP_DISTANCES and not GL_MAX_CLIP_PLANES in our gl headers so I used that

Clip planes were fixed function, now I guess they're called distances, but the HW limits are identical.